### PR TITLE
add docker-compose

### PR DESCRIPTION
**Pull request by dpfaffenbauer**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+version: '2.0'
```
Comment (yariksheptykin) [inline docker-compose.yml]: I think we could go up to version `2.4`  with this file.

```suggestion
version: '2.4'
```

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    image: pimcore/pimcore:PHP7.4-apache
+    container_name: pimcore-php
+    volumes:
+      - .:/var/www/html:cached
+    ports:
+     - 80:80
+     - 443:443
+    links:
+     - db:db
+     - redis:redis
```
Comment (yariksheptykin) [inline docker-compose.yml]: I believe we can safely drop this. compse should make services via their names available via DNS resolution.

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  php-debug:
+    image: pimcore/pimcore:PHP7.4-apache-debug
+    container_name: pimcore-debug-php
+    volumes:
+      - .:/var/www/html:cached
+    ports:
+     - 88:80
+    links:
+     - db:db
+     - redis:redis
```
Comment (yariksheptykin) [inline docker-compose.yml]: I believe we can safely drop this. compse should make services via their names available via DNS resolution.

File: docker-compose.yml (hunk 2 above)
Comment (dpfaffenbauer) [reply docker-compose.yml]: done

File: docker-compose.yml (hunk 3 above)
Comment (dpfaffenbauer) [reply docker-compose.yml]: done

**Comment by yariksheptykin** [comment]
Thanks for contributing the composer file, @dpfaffenbauer ! Donating your images to pimcore is a great idea!

**Comment by dpfaffenbauer** [comment]
@yariksheptykin thanks. I told you, good things take time ;)

File: README.md [hunk 4]
```diff
@@ -14,5 +14,17 @@ cd ./my-project
...
 - Done! 😎
 
+## Docker
+
+You can also use Docker to setup a new Pimcore Installation:
+
+```bash
+COMPOSER_MEMORY_LIMIT=-1 composer create-project pimcore/skeleton my-project
+cd ./my-project
+docker-compose exec php vendor/bin/pimcore-install --mysql-host-socket=db
```
Comment (yariksheptykin) [inline README.md]:
```suggestion
docker-compose run --rm php vendor/bin/pimcore-install --mysql-host-socket=db
docker-compose run --rm php chown -R www-data:www-data var/*
docker-compose up -d
```

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+      - 8080:8080
+
+  php:
+    image: pimcore/pimcore:PHP7.4-apache
+    container_name: pimcore-php
+    volumes:
+      - .:/var/www/html:cached
+    ports:
+     - 80:80
+     - 443:443
```
Comment (yariksheptykin) [inline docker-compose.yml]: This allows running one-off php containers with `docker-compose run --rm php ...` that depend on db such as installing pimcore for example.

```suggestion
     - 443:443
    depends_on:
     - db
```

File: docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+  php-debug:
+    image: pimcore/pimcore:PHP7.4-apache-debug
```
Comment (yariksheptykin) [inline docker-compose.yml]: `PHP7.4-apache-debug` does not seem to be existing.

```
docker pull pimcore/pimcore:PHP7.4-apache-debug
Error response from daemon: manifest for pimcore/pimcore:PHP7.4-apache-debug not found: manifest unknown: manifest unknown
```
We could downgrade to 7.3 for debug image for now:

```suggestion
    image: pimcore/pimcore:PHP7.3-apache-debug
```

File: README.md [hunk 7]
```diff
@@ -14,5 +14,17 @@ cd ./my-project
...
+```
+
```
Comment (yariksheptykin) [inline README.md]:
```suggestion
You can now navigate your browser to https://localhost or https://localhost/admin.
```

File: docker-compose.yml (hunk 6 above)
Comment (dpfaffenbauer) [reply docker-compose.yml]: Images currently aren't working yet correctly, should work tomorrow (building them takes like forever)

Comment (yariksheptykin) [reply docker-compose.yml]: I can confirm that PHP7.4-apache-debug is available now:

```
docker pull pimcore/pimcore:PHP7.4-apache-debug
PHP7.4-apache-debug: Pulling from pimcore/pimcore
000eee12ec04: Already exists 
8ae4f9fcfeea: Already exists 
60f22fbbd07a: Already exists 
ccc7a63ad75f: Already exists 
a2427b8dd6e7: Already exists 
91cac3b30184: Already exists 
d6e40015fc10: Already exists 
9858aa646efe: Already exists 
7940985f7eb2: Already exists 
b23f72eebcfb: Already exists 
75bb7b8d192c: Already exists 
7edf943992b0: Already exists 
c8bf9d9d0e11: Already exists 
22e67553f831: Pull complete 
1d659fd986be: Pull complete 
1eec1df56135: Pull complete 
80dbf54decea: Pull complete 
fb6904e26091: Pull complete 
dc00a49875d1: Pull complete 
657c1f64e874: Pull complete 
57740df64ce1: Pull complete 
de0091d9d3b2: Pull complete 
Digest: sha256:ff6aafd19113656049bc7210e148236130a59b9885f0aced540a0f5d443f6f5f
Status: Downloaded newer image for pimcore/pimcore:PHP7.4-apache-debug
docker.io/pimcore/pimcore:PHP7.4-apache-debug
```

**Comment by brusch** [comment]
Thanks a lot! 🙌

**Comment by yariksheptykin** [comment]
Happy to see this landed. Thanks @dpfaffenbauer for your patience.

**Pull request merged**